source/language/standard_library.rst

```diff
@@ -0,0 +1,329 @@
+Standard library
+================
+
+The standard library for OpenQASM 3 consists of a single include file, called ``stdgates.inc``.
+This file can be included in an OpenQASM 3 program with the statement
+
+.. code-block::
+
+   include "stdgates.inc";
+
+Versioning
+----------
+
+The standard library is versioned directly with the language itself.  This documentation will note
+the version that any given gate in the standard library became available, and, if applicable, the
+version it was removed in.
+
+Notes for implementations
+-------------------------
+
+An implementation is not required to make the standard library available in situations where the
+OpenQASM 3 program is being interpreted in terms of the instruction set architecture of a particular
+QPU.  The rest of this section concerns only situations where the library is considered available.
+
+Implementations should consider the ``stdgates.inc`` file to be locatable, regardless of any other
+settings that they might apply around include search paths.
+
+Upon interpreting an ``include "stdgates.inc";`` statement, implementations should make available
+exactly the set of gates defined in the standard library for the language version that matches the
```

Comment:
There'd been some worry from people that if the file had versions, it would be a bit of a nuance to manage if all implementations actually had to parse a real file backing it, since the compatibility version of the parse might only be determined dynamically when the `OPENQASM 3.x` statement is encountered..  The point of the provision was to make it clear that as long as an implementation behaves _as if_ there was a real file backing it, that's fine.